### PR TITLE
Create more idiomatic API for .NET

### PR DESCRIPTION
**Pull request by lokitoth**
* Adds non-TryXYZ equivalents which wrap the ApiStatus mechanism into a proper exception
* Fixes an issue where eventId was incorrectly mislabeled as actionId (#39)

### CODE REVIEW

File: bindings/cs/rl.net/RLException.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            this.ErrorCode = status.ErrorCode;
+        }
+
+        protected RLException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            this.ErrorCode = info.GetInt32(ErrorCodeSerializationKey);
+        }
+
+        [SecurityPermission(SecurityAction.Demand, SerializationFormatter = true)]
```
Comment (kumpera) [inline bindings/cs/rl.net/RLException.cs]: Do we still need this stuff?

File: bindings/cs/rl.net/LiveModel.cs [hunk 2]
```diff
@@ -72,48 +72,108 @@ public bool TryInit(ApiStatus apiStatus = null)
...
+        public bool TryReportActionTaken(string eventId, ApiStatus apiStatus = null)
+        {
+            int result = LiveModelReportActionTaken(this.NativeHandle, eventId, apiStatus.ToNativeHandleOrNullptr());
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryReportOutcome(string actionId, float outcome, ApiStatus apiStatus = null)
+        public void ReportActionTaken(string eventId)
+        {
+            ApiStatus apiStatus = new ApiStatus();
```
Comment (kumpera) [inline bindings/cs/rl.net/LiveModel.cs]: We should be disposing the temporary ApiStatus objects.

File: bindings/cs/rl.net/RLException.cs (hunk 1 above)
Comment (lokitoth) [reply bindings/cs/rl.net/RLException.cs]: If we want to support scenarios like remoting (or serializing for some other reason) exceptions, we need this. Is there a good reason to remove it?

File: bindings/cs/rl.net/LiveModel.cs (hunk 2 above)
Comment (lokitoth) [reply bindings/cs/rl.net/LiveModel.cs]: Good call.

**Comment by lokitoth** [comment]
Simple rebase on top of master

File: bindings/cs/rl.net/RLException.cs (hunk 1 above)
Comment (kumpera) [reply bindings/cs/rl.net/RLException.cs]: dotnet code doesn't support remoting, but has most of the serialization stack.
It's harmless to have it, OTOH.

Comment (lokitoth) [reply bindings/cs/rl.net/RLException.cs]: Fair enough. I'll keep this since we technically do support .Net Framework.

**Comment by lokitoth** [comment]
Will be merging this in EOD today. Please let me know if there are any more comments/suggestions.

**Pull request merged**